src/main/java/org/apache/maven/plugins/dependency/GetClassesMojo.java

```diff
@@ -0,0 +1,406 @@
+        repositorySystem.injectAuthentication( repoList, settings.getServers() );
+
+        buildingRequest.setRemoteRepositories( repoList );
+
+        return buildingRequest;
+    }
+
+    protected ArtifactCoordinate toArtifactCoordinate( DependableCoordinate dependableCoordinate )
+    {
+        ArtifactHandler artifactHandler = artifactHandlerManager.getArtifactHandler( dependableCoordinate.getType() );
+        DefaultArtifactCoordinate artifactCoordinate = new DefaultArtifactCoordinate();
+        artifactCoordinate.setGroupId( dependableCoordinate.getGroupId() );
+        artifactCoordinate.setArtifactId( dependableCoordinate.getArtifactId() );
+        artifactCoordinate.setVersion( dependableCoordinate.getVersion() );
+        artifactCoordinate.setClassifier( dependableCoordinate.getClassifier() );
+        artifactCoordinate.setExtension( artifactHandler.getExtension() );
+        return artifactCoordinate;
+    }
+
+    ArtifactRepository parseRepository( String repo, ArtifactRepositoryPolicy policy )
+            throws MojoFailureException
+    {
+        // if it's a simple url
+        String id = "temp";
+        ArtifactRepositoryLayout layout = getLayout( "default" );
+        String url = repo;
+
+        // if it's an extended repo URL of the form id::layout::url
+        if ( repo.contains( "::" ) )
+        {
+            Matcher matcher = ALT_REPO_SYNTAX_PATTERN.matcher( repo );
+            if ( !matcher.matches() )
+            {
+                throw new MojoFailureException( repo, "Invalid syntax for repository: " + repo,
+                        "Invalid syntax for repository. Use \"id::layout::url\" or \"URL\"." );
+            }
+
+            id = matcher.group( 1 ).trim();
+            if ( !StringUtils.isEmpty( matcher.group( 2 ) ) )
+            {
+                layout = getLayout( matcher.group( 2 ).trim() );
+            }
+            url = matcher.group( 3 ).trim();
+        }
+        return new MavenArtifactRepository( id, url, layout, policy, policy );
+    }
+
+    private ArtifactRepositoryLayout getLayout( String id )
+            throws MojoFailureException
+    {
+        ArtifactRepositoryLayout layout = repositoryLayouts.get( id );
+
+        if ( layout == null )
+        {
+            throw new MojoFailureException( id, "Invalid repository layout", "Invalid repository layout: " + id );
+        }
+
+        return layout;
+    }
+
+    /**
+     * @return {@link #skip}
+     */
+    protected boolean isSkip()
+    {
+        return skip;
+    }
+
+    protected boolean isTransitive()
+    {
+        return transitive;
+    }
+
+    protected DefaultDependableCoordinate getCoordinate()
+    {
+        return coordinate;
+    }
+
+    protected DependencyResolver getDependencyResolver()
+    {
+        return dependencyResolver;
+    }
+
+    protected ArtifactResolver getArtifactResolver()
+    {
+        return artifactResolver;
+    }
+
+    /**
+     * @param artifact The artifact
```

Comment:
doc comments begin with lower case, per Oracle guidelines

Reply:
More clarification? This is taken from the GetMojo.